lib/hooks/index.js

```diff
@@ -97,25 +67,51 @@ function findModuleVersion(request, parent, load) {
     if (mainScriptDir.indexOf(paths[i]) === 0) {
       // This is unsafe on windows if request contains a namespace
       // TODO: convert path separators in request
-      var pjson = load(path.join(paths[i], request, 'package.json'));
-      return pjson.version;
+      return path.join(paths[i], request);
     }
   }
   return null;
 }
 
+/**
+ * Determines the version of the module located at `modulePath`.
+ *
+ * @param {?string} modulePath The absolute path to the root directory of the
+ *    module being loaded. This may be null if we are loading an internal module
+ *    such as http.
+ */
+function findModuleVersion(modulePath, load) {
+  return modulePath ? load(path.join(modulePath, 'package.json')).version :
```

Comment:
Sorry, I'll be less obtuse this time. 

> earlier today we agreed that we already do not patch non-npm installed modules at all

That's not what we agreed. My claim is that you can't load a module specified by name (as opposed to a relative or absolute path) unless it exists inside some `node_modules/${package_name}`. That `node_modules` needs to be in the module search path for the parent module.

I think it is possible for a user to self-install a directory inside `node_modules`. If that directory doesn't contain a `package.json`, the code above would crash.

It should never crash.